src/main/java/org/casbin/jcasbin/main/CoreEnforcer.java

```diff
@@ -339,8 +381,10 @@ public void enableAutoBuildRoleLinks(boolean autoBuildRoleLinks) {
      * role inheritance relations.
      */
     public void buildRoleLinks() {
-        rm.clear();
-        model.buildRoleLinks(rm);
+        for (Map.Entry<String, RoleManager> entry : rmMap.entrySet()) {
```

Comment:
you could iterate over `rmMap.values()` instead of `rmMap.entrySet()` here to save yourself the `entry.getValue()` calls

Reply:
Yes, you are right. By iterating over `rmMap.entrySet()` is more efficient when processing large amounts of data, but this place should handle very little data, so it should use `rmMap.values()`. I will finish it later.